packages/kit/types/index.d.ts

```diff
@@ -71,6 +71,22 @@ type UnpackValidationError<T> = T extends ActionFailure<infer X>
 	? undefined // needs to be undefined, because void will corrupt union type
 	: T;
 
+export type ActionsSuccess<T extends Record<string, (...args: any) => any>> = {
+	[Key in keyof T]: ExcludeValidationError<Awaited<ReturnType<T[Key]>>>;
+}[keyof T];
+
+type ExcludeValidationError<T> = T extends ActionFailure<any> ? never : T extends void ? never : T;
+
+export type ActionsInvalid<T extends Record<string, (...args: any) => any>> = {
```

Comment:
`ActionsFailure` would be the more consistent name

Reply:
About consistency ^^
![image](https://user-images.githubusercontent.com/21335119/223679382-182cc50b-6a41-41d3-bbde-f0f696fec700.png)

I have chosen `Invalid` because the `SubmitFunction` uses it as the name for the generic.
And there already exists a type `ActionFailure` so `ActionsFailure` could be confusing.

What should we do?

Reply:
Oh, that's an oversight I'd say from the days where the method for the actions was called `invalid` - let's rename that generic to `Failure`.

Reply:
So I rename `Invalid` to `Failure` and `ActionsInvalid` to `ActionsFailure`, right?